imports/wasi_snapshot_preview1/example/testdata/zig/cat.zig

```diff
@@ -0,0 +1,41 @@
+const std = @import("std");
+const io = std.io;
+const os = std.os;
+const allocator = std.heap.page_allocator;
+const PreopenList = std.fs.wasi.PreopenList;
+const stdout = std.io.getStdOut();
+const warn = std.log.warn;
+
+pub fn main() !void {
```

Comment:
cc @Luukdegram @knqyf263 @mathetake for wasi+zig style hints I found pre-opens the most unintuitive as I figured "/" would somehow be easier to map

Reply:
I expected that we don't need to access WASi specifics like PreopenList...

Reply:
ok so I imagine the reason behind this is Zig stdlib maintains the preopens in memory and therefore it requires the allocation, and it turns out that we need to call WASI specific APIs in the main to avoid hidden allocations. sounds plausible

Reply:
yeah I should clarify the best comment and cite https://ziglang.org/documentation/0.10.0/#WASI

Reply:
still feels off that we can't avoid adding wasi-specific code

Reply:
added comment